File: src/main/java/com/itmo/microservices/demo/payment/api/service/PaymentService.java
package com.itmo.microservices.demo.payment.api.service;

import com.itmo.microservices.demo.payment.api.model.PaymentSubmissionDto;
import com.itmo.microservices.demo.payment.api.model.UserAccountFinancialLogRecordDto;
import com.itmo.microservices.demo.users.api.exception.UserNotFoundException;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.List;
import java.util.UUID;

public interface PaymentService {

    List<UserAccountFinancialLogRecordDto> getFinlog(String username, UUID orderId) throws UserNotFoundException;
    PaymentSubmissionDto executeOrderPayment(UserDetails user, UUID orderId);
}

File: src/main/kotlin/com/itmo/microservices/demo/payment/api/service/PaymentService.kt
package com.itmo.microservices.demo.payment.api.service

import com.itmo.microservices.demo.payment.api.model.PaymentSubmissionDto
import com.itmo.microservices.demo.payment.api.model.UserAccountFinancialLogRecordDto
import java.util.UUID

interface PaymentService {
    fun getPaymentFinLog(order_id: UUID): List<UserAccountFinancialLogRecordDto>

    fun orderPayment(order_id: UUID): PaymentSubmissionDto
}